x/lockup/keeper/lock_refs.go

```diff
@@ -6,7 +6,10 @@ import (
 )
 
 func (k Keeper) addLockRefs(ctx sdk.Context, lock types.PeriodLock) error {
-	refKeys, err := lockRefKeys(lock)
+	refKeys, err := durationLockRefKeys(lock)
+	if lock.IsUnlocking() {
+		refKeys, err = lockRefKeys(lock)
```

Comment:
imo, I would prefer using something like `unlockingRefKeys` instead of `lockReftKeys` but its a minor thing, totally up to you!